src/treestatus/api/treestatus_api/api.py

```diff
@@ -36,7 +51,184 @@ def _now():
+        headers=headers,
+        json=_statuspage_data(False,
+                              component_id,
+                              tree,
+                              status_from,
+                              status_to,
+                              ),
+    )
+    try:
+        response.raise_for_status()
+    except Exception:
+        _statuspage_send_email_on_error(
+            subject=f'[treestatus] Error when creating statuspage incident',
+            content=STATUSPAGE_ERROR_ON_CREATE.format(tree=tree.tree),
+            link=dict(
+                href=f'https://manage.statuspage.io/pages/{page_id}',
+                text='Visit statuspage',
+            ),
+        )
+        pass
+
+    return response.json()
+
+
+def _close_statuspage_incident(
+    headers,
+    component_id,
+    tree,
+    status_from,
+    status_to,
+):
+    page_id = flask.current_app.config.get('STATUSPAGE_PAGE_ID')
+    response = requests.get(
+        f'{STATUSPAGE_URL}/pages/{page_id}/incidents/unresolved',
+        headers=headers,
+    )
+    try:
+        response.raise_for_status()
+    except Exception:
+        _statuspage_send_email_on_error(
+            subject=f'[treestatus] Error when closing statuspage incident',
+            content=STATUSPAGE_ERROR_ON_CLOSE.format(tree=tree.tree),
+        )
+
+    # last incident with meta.treestatus.tree == tree.tree
+    incident_id = None
+    for incident in sorted(response.json(), key=lambda x: x['created_at']):
+        if 'metadata' in incident and \
+                'treestatus' in incident['metadata'] and \
+                'tree' in incident['metadata']['treestatus'] and \
+                incident['metadata']['treestatus'] == tree.tree:
+            incident_id = incident['id']
+            break
+
+    if incident_id is None:
+        _statuspage_send_email_on_error(
+            subject=f'[treestatus] Error when closing statuspage incident',
+            content=STATUSPAGE_ERROR_ON_CLOSE.format(tree=tree.tree),
+        )
+
+    response = requests.patch(
+        f'{STATUSPAGE_URL}/pages/{page_id}/incidents/{incident_id}',
+        headers=headers,
+        json=_statuspage_data(True,
+                              component_id,
+                              tree,
+                              status_from,
+                              status_to,
+                              ),
+    )
+    try:
+        response.raise_for_status()
+    except Exception:
+        _statuspage_send_email_on_error(
+            subject=f'[treestatus] Error when closing statuspage incident',
+            content=STATUSPAGE_ERROR_ON_CLOSE.format(tree=tree.tree),
+            incident_id=incident_id,
+        )
+
+    return response.json()
+
+
 def _notify_status_change(trees_changes, tags=[]):
+    if flask.current_app.config.get('STATUSPAGE_ENABLE'):
+        components = flask.current_app.config.get('STATUSPAGE_COMPONENTS')
+        token = flask.current_app.config.get('STATUSPAGE_TOKEN')
```

Comment:
Looks like `STATUSPAGE_COMPONENTS` and `STATUSPAGE_TOKEN` are required config variables, but it's not enforced (in `settings.py`). Should we make sure that they exist before using them, so we don't try using `None`?